packages/account/src/providers/provider.ts

```diff
@@ -1635,4 +1636,15 @@ export default class Provider {
   async getTransactionResponse(transactionId: string): Promise<TransactionResponse> {
     return new TransactionResponse(transactionId, this);
   }
+
+  /**
+   * Returns Message for given nonce.
+   *
+   * @param nonce - The nonce of the message to retrieve.
+   * @returns A promise that resolves to the Message object.
+   */
+  async getMessageByNonce(nonce: string): Promise<GqlMessage | undefined | null> {
```

Comment:
Should we just return a `GqlMessage` with the values set to a default as opposed to a `undefined` | `null` ? It's quite difficult to work with APIs that could return `null` | `undefined` as well as a type

Reply:
@maschad I don't think we should or we can fill in defaults since that would be unexpected behaviour. I think we should simplify the API though. I will return `undefined` in case there's no message found, and the actual message if we find something.